src/data.rs

```diff
@@ -1,142 +1,152 @@
-        // Big Endian byte order stores low order bytes last.
         let mut result: u32 = 0;
         for (i, byte) in bytes.iter().enumerate() {
-            // println!("Big End Byte: {}", byte);
             result |= (*byte as u32) << (bits * (bytes.len() - i - 1));
-            // println!("Big End Result: {}", result);
         }
         result
     }
 
-    pub fn default_decode(byte_vals: &[u8]) -> Vec<i64> {
-        // Default decode structure seen by a majority of the messages.
-
-        let grouped_vals = ProcessData::group_bytes(byte_vals, 2);
-        println!("CUCKED GROUP BYTES");
-        let parsed_vals: Vec<u32> = grouped_vals
-            .iter()
-            .map(|val| ProcessData::little_endian(val, 8))
-            .collect();
-        println!("CUCKED LITTLE ENDIAN");
-        let decoded_vals: Vec<i64> = parsed_vals
-            .iter()
-            .map(|val| ProcessData::twos_comp(*val, 16))
-            .collect();
-        println!("CUCKED TWOS COMP");
-        decoded_vals
-    }
+    /**
+     * Decodes the given byte by taking the top four bits after shifting it by the given number of bits.
+     */
+    pub fn half(byte: u8, bits: u8) -> u32 {
+        (byte >> bits & 15) as u32
+    } 
 }
 
-pub struct FormatData {
-    // Utility functions to scale data values of a specific type.
-}
+/**
+ * Class to contain the data formatting functions
+ */
+pub struct FormatData {}
 
 impl FormatData {
-    pub fn temperature(value: i64) -> f32 {
-        value as f32 / 10.0
+    pub fn temperature(value: f32) -> f32 {
+        value / 10.0
     }
 
-    pub fn low_voltage(value: i64) -> f32 {
-        value as f32 / 100.0
+    pub fn low_voltage(value: f32) -> f32 {
+        value / 100.0
     }
 
-    pub fn torque(value: i64) -> f32 {
-        value as f32 / 10.0
+    pub fn torque(value: f32) -> f32 {
+        value / 10.0
     }
 
-    pub fn high_voltage(value: i64) -> f32 {
-        value as f32 / 10.0
+    pub fn high_voltage(value: f32) -> f32 {
+        value / 10.0
     }
 
-    pub fn current(value: i64) -> f32 {
-        value as f32 / 10.0
+    pub fn current(value: f32) -> f32 {
+        value / 10.0
     }
 
-    pub fn angle(value: i64) -> f32 {
-        value as f32 / 10.0
+    pub fn angle(value: f32) -> f32 {
+        value / 10.0
     }
 
-    pub fn angular_velocity(value: i64) -> i64 {
+    pub fn angular_velocity(value: f32) -> f32 {
         -value
     }
 
-    pub fn frequency(value: i64) -> f32 {
-        value as f32 / 10.0
+    pub fn frequency(value: f32) -> f32 {
+        value / 10.0
+    }
+
+    pub fn power(value: f32) -> f32 {
+        value / 10.0
+    }
+
+    pub fn timer(value: f32) -> f32 {
+        value * 0.003
     }
 
-    pub fn power(value: i32) -> f32 {
-        value as f32 / 10.0
+    pub fn flux(value: f32) -> f32 {
+        value / 1000.0
     }
 
-    pub fn timer(value: i32) -> f32 {
-        value as f32 * 0.003
+    pub fn cell_voltage(value: f32) -> f32 {
```

Comment:
Why do we need a different format for cell voltage? Just because we broadcast with a higher resolution and need to normalize differently?

Reply:
Yeah its dividing by 1000 whe i think the other voltage formats are only by ten